dockerfile_parse/parser.py

```diff
@@ -871,10 +871,12 @@ def image_from(from_value):
     :param from_value: string like "image:tag" or "image:tag AS name"
     :return: tuple of the image and stage name, e.g. ("image:tag", None)
     """
-    regex = re.compile(r"""(?xi)     # readable, case-insensitive regex
-        \s*                          # ignore leading whitespace
-        (?P<image> \S+ )             # image and optional tag
-        (?:                          # optional "AS name" clause for stage
+    regex = re.compile(r"""(?xi)        # readable, case-insensitive regex
+        \s*                             # ignore leading whitespace
+        (?P<platform> --platform=\S+)?  # optional platform parameter
```

Comment:
can you use just ? : 
(?: --platform=\S+)?

since 'platform' group isn't used anyway

Reply:
Of ocurse it's possible: But I would think that in the future you would also want to allow to edit this platform parameter?

The current PR is just a (non-breaking) bugfix. But Ideally I think you guys should concider if this "image_from" approach is the way to go (and shouldn't be more of a "parse_FROM_keyword()" function which returns all values that are allowed within a FROM keyword.

See my other [PR](https://github.com/containerbuildsystem/dockerfile-parse/pull/129) on why I personally think that might be a good idea.